### PR TITLE
Make WCS creation and header creation more lenient

### PR DESCRIPTION
**Pull request by dstansby**
Update of #3628 - @jmason86 is there any chance you could post the fits header that was causing problems, so we can include it as a test?

### CODE REVIEW

**Comment by jmason86** [comment]
Is the fits header and the `.meta` basically the same thing? If so, I can give you a quick answer [here](https://github.com/sunpy/sunpy/issues/3248#issuecomment-568008359). Else, it'll be a few days before I can track down what I did (at AAS presently).

**Comment by dstansby** [comment]
Yep, the `.meta` should be fine, thanks a lot!

**Comment by Cadair** [comment]
here is a header:
```
# HDU 0 in /home/stuart/sunpy/data/archive_soho_private_data_processed_lasco_level_05_level_05_150618_c2_22551989.fts:
SIMPLE  =                    T / Written by IDL:  Fri Oct 30 01:38:26 2015      
BITPIX  =                   16 /                                                
NAXIS   =                    2 /                                                
NAXIS1  =                 1024 /                                                
NAXIS2  =                 1024 /                                                
FILENAME= '22551989.fts'       /                                                
FILEORIG= '150618_000207.img'  /                                                
DATE    = '2015/10/30 05:38:23.081' /                                           
DATE-OBS= '2015/06/18'         /                                                
TIME-OBS= '00:00:05.794'       /                                                
P1COL   =                   20 /                                                
P1ROW   =                    1 /                                                
P2COL   =                 1043 /                                                
P2ROW   =                 1024 /                                                
VERSION =                    2 /                                                
EXPTIME =              25.0977 /                                                
EXP0    =              23.0000 /                                                
EXPCMD  =              23.0000 /                                                
EXP1    =              1.88428 /                                                
EXP2    =              2.37305 /                                                
EXP3    =              2.09766 /                                                
TELESCOP= 'SOHO    '           /                                                
INSTRUME= 'LASCO   '           /                                                
DETECTOR= 'C2      '           /                                                
READPORT= 'C       '           /                                                
SUMROW  =                    0 /                                                
SUMCOL  =                    0 /                                                
LEBXSUM =                    1 /                                                
LEBYSUM =                    1 /                                                
SHUTTR  =                    0 /                                                
LAMP    =                    0 /                                                
FILTER  = 'Orange  '           /                                                
POLAR   = 'Clear   '           /                                                
LP_NUM  = 'Normal  '           /                                                
OS_NUM  =                 4136 /                                                
IMGCTR  =                 1622 /                                                
IMGSEQ  =                    0 /                                                
COMPRSSN= 'XZ%     '           /                                                
HCOMP_SF=                   64 /                                                
MID_DATE=                57191 /                                                
MID_TIME=              18.3428 /                                                
PLATESCL=              11.9000 /                                                
OFFSET  =              631.118 /                                                
IMAGE_CT=                 1622 /                                                
SEQ_NUM =                    0 /                                                
OBT_TIME=        1813276840.79 /                                                
R1COL   =                   20 /                                                
R1ROW   =                    1 /                                                
R2COL   =                 1043 /                                                
R2ROW   =                 1024 /                                                
EFFPORT = 'C       '           /                                                
RECTIFY = 'TRUE    '           /                                                
DATAMIN =              633.000 /                                                
DATAMAX =              16384.0 /                                                
DATAZER =                    0 /                                                
DATASAT =                    0 /                                                
DATAAVG =              2444.65 /                                                
DATASIG =              1164.91 /                                                
DATAP01 =                  710 /                                                
DATAP10 =                 1192 /                                                
DATAP25 =                 1804 /                                                
DATAP75 =                 2938 /                                                
DATAP90 =                 3496 /                                                
DATAP95 =                 3900 /                                                
DATAP98 =                 4782 /                                                
DATAP99 =                 7476 /                                                
CRPIX1  =              511.200 /; Using SOHO roll + offset and suncenter estimat
CRPIX2  =              507.500 /                                                
CRVAL1  =              0.00000 /                                                
CRVAL2  =              0.00000 /                                                
CROTA1  =              7.62989 /; Using SOHO roll + offset and suncenter estimat
CROTA2  =              7.62989 /                                                
CTYPE1  = 'SOLAR-X '           /                                                
CTYPE2  = 'SOLAR-Y '           /                                                
CUNIT1  = 'ARCSEC  '           /                                                
CUNIT2  = 'ARCSEC  '           /                                                
CDELT1  =              11.9000 /                                                
CDELT2  =              11.9000 /                                                
HISTORY offset_bias.pro 1.37, 12/18/14, 631.118                                 
HISTORY V15 18 Jun 1997 MAKE_FITS_HDR                                           
HISTORY 2  
```

**Comment by dstansby** [comment]
I decided to go minimal for the test...

**Comment by dstansby** [comment]
If you could that would be great!

**Comment by nabobalis** [comment]
DO we not want to add that header that stuart gave as a test (does not have to be here)?

File: changelog/3645.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Correctly catch and emit a warning when a .fits file contains a key word with
```
Comment (Cadair) [inline changelog/3645.bugfix.rst]:
```suggestion
Correctly catch and emit a warning when converting a map metadata to a FITS header and it contains a keyword with
```

**Comment by wtbarnes** [comment]
I think David's minimal headers still provide an adequate test. The only problem with that other header was the tab character.

File: changelog/3645.bugfix.rst (hunk 1 above)
Comment (wtbarnes) [reply changelog/3645.bugfix.rst]: Stupid GH suggestions. Fixed manually.

File: changelog/3645.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Correctly catch and emit a warning when converting a map metadata to a FITS
```
Comment (Cadair) [inline changelog/3645.bugfix.rst]: Sorry:
```suggestion
Correctly catch and emit a warning when converting map metadata to a FITS
```

Comment (wtbarnes) [reply changelog/3645.bugfix.rst]: Too late 🤣

**Pull request merged**